### ISSUE TITLE
React-elastic-carousel only functional after resizing window

### ISSUE THREAD

**Issue by pixldeepak** [sag1v/react-elastic-carousel#164]
I've just started using the react-elastic-carousel package and am running into an issue on page load. Everything loads up on page load/refresh, but when I click the arrow to move the carousel, the items aren't rotating. Only upon resizing the window do the products re-format correctly, and then I can continue to use the buttons normally from then on. I am attaching the link here. [https://stackoverflow.com/questions/66487320/react-elastic-carousel-only-functional-after-resizing-window](url)

**Comment by sag1v** [comment]
Are you doing server side rendering? E.g gatsby or similar?

**Comment by pixldeepak** [comment]
Yes i am doing server side rendering

**Comment by pejman1** [comment]
> Are you doing server side rendering? E.g gatsby or similar?

I have same problem when I use concurrent mode in react and package is not compatible with concurrent mode

**Comment by sag1v** [comment]
Check [this comment](https://github.com/sag1v/react-elastic-carousel/issues/130#issuecomment-749340515)

TLDR; 
Try using `gatsby-plugin-styled-components`

**Comment by pejman1** [comment]
> Check [this comment](https://github.com/sag1v/react-elastic-carousel/issues/130#issuecomment-749340515)
> 
> TLDR;
> Try using `gatsby-plugin-styled-components`

How about the issue in concurrent mode. can you fix it?

**Comment by sag1v** [comment]
What exactly is the problem, is it related to this issue? If not, maybe its better to open a seperate issue.

**Comment by pejman1** [comment]
> What exactly is the problem, is it related to this issue? If not, maybe its better to open a seperate issue.

It is exactly same issue

**Comment by sag1v** [comment]
So are you using gatsby and have you tried `gatsby-plugin-styled-components`?

**Comment by pejman1** [comment]
> So are you using gatsby and have you tried `gatsby-plugin-styled-components`?

No, I am using react in electron. the problem is related to concurrent mode in react not server side rendering 
if you enable concurrent mode you face same issue

**Comment by pejman1** [comment]
> So are you using gatsby and have you tried `gatsby-plugin-styled-components`?

No, I am using react in electron. the problem is related to concurrent mode in react not server side rendering 
if you enable concurrent mode you face same issue

**Comment by sag1v** [comment]
Ok then, its not the exact same issue.
If you think its related to concurrent mode, can you open a different issue with an example in codesandbox.io ?

**Comment by pejman1** [comment]
> Ok then, its not the exact same issue.
> If you think its related to concurrent mode, can you open a different issue with an example in codesandbox.io ?

my answer was clear, if you enable concurrent mode react - Everything loads up on page load/refresh, but when you click the arrow to move the carousel, the items aren't rotating. Only upon resizing the window do the products re-format correctly, and then you can continue to use the buttons normally from then on.

**Comment by sag1v** [comment]
@pixldeepak I think your issue might be related to [this issue](https://github.com/sag1v/react-elastic-carousel/issues/163#issuecomment-808724065)

Try installing latest version (`0.11.3`) 

Let me know if that worked out for you.

**Comment by sag1v** [comment]
@pejman1 I think your issue is not related to concurrent mode at all. I guess you just had an empty array intially and then added items (like the OP had). 

Here is a working example with concurrent mode
https://codesandbox.io/s/react-elastic-carousel-forked-zy43u?file=/src/index.js:406-428

**Comment by pejman1** [comment]
> @pejman1 I think your issue is not related to concurrent mode at all. I guess you just had an empty array intially and then added items (like the OP had).
> 
> Here is a working example with concurrent mode
> https://codesandbox.io/s/react-elastic-carousel-forked-zy43u?file=/src/index.js:406-428

Thank you so much. I updated to the last version and the problem is gone.

**Comment by sag1v** [comment]
Can you reproduce this? i tried and i don't get this behavior

**Comment by pejman1** [comment]
> Can you reproduce this? i tried and i don't get this behavior

it is in an application . I load data by pouchdb (loading data asynchronies ) and then load it in carousel but in intial load it sometimes does not show anything until I resize the window .  to solve the problem I refresh and rerender the component after 300ms

**Comment by deepakmalik264** [comment]
I am also facing the same problem

**Comment by sag1v** [comment]
@deepakmalik264 / @pejman1 Sorry but i can't reproduce this behavior, without an example i can't figure out what's wrong and can't fix it.

BTW, is there a special reason why you render a the Carousel with no items? I would just conditionally render it only when i have the items ready, i mean the Carousel with no items is kind of pointless (from a user's perspective)

**Comment by pejman1** [comment]
> @deepakmalik264 / @pejman1 Sorry but i can't reproduce this behavior, without an example i can't figure out what's wrong and can't fix it.
> 
> BTW, is there a special reason why you render the Carousel with no items? I would just conditionally render it only when i have the items ready, i mean the Carousel with no items is kind of pointless (from a user's perspective)
I don't render it with no Item 
I use suspense feedback and don't render component before data is ready. but in concurrent mode it doesn't show items sometimes before re-rendering  the component

**Comment by sag1v** [comment]
Like i said, i need an example of the behavior as i can't reproduce it.
